routes/extra_data_encoding.go

```diff
@@ -0,0 +1,200 @@
+}
+
+// Encode64BitIntString supports encoding integers up to a length of 8 bytes
+func Encode64BitIntString(str string) ([]byte, error) {
+	var encoded, err = strconv.ParseInt(str, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	var buffer []byte
+	lib.PutVarint(buffer, encoded)
+	return buffer, nil
+}
+
+// Decode64BitUintString supports decoding unsigned integers up to a length of 8 bytes
+func Decode64BitUintString(bytes []byte, _ *lib.DeSoParams, _ *lib.UtxoView) string {
+	var decoded, _ = lib.Uvarint(bytes)
+	return strconv.FormatUint(decoded, 10)
+}
+
+// Encode64BitUintString supports decoding unsigned integers up to a length of 8 bytes
+func Encode64BitUintString(str string) ([]byte, error) {
+	var encoded, err = strconv.ParseUint(str, 10, 64)
+	if err != nil {
+		return nil, err
+	}
+	var buffer []byte
+	lib.PutUvarint(buffer, encoded)
+	return buffer, nil
+}
+
+func DecodeBoolString(bytes []byte, params *lib.DeSoParams, utxoView *lib.UtxoView) string {
+	return Decode64BitUintString(bytes, params, utxoView)
+}
+
+func EncodeBoolString(str string) ([]byte, error) {
+	return Encode64BitUintString(str)
+}
+
+func DecodeHexString(bytes []byte, _ *lib.DeSoParams, _ *lib.UtxoView) string {
+	return hex.EncodeToString(bytes)
+}
+
+func EncodeHexString(str string) ([]byte, error) {
+	return hex.DecodeString(str)
+}
+
+func EncodePkStringToBytes(str string) ([]byte, error) {
+	result, _, err := lib.Base58CheckDecode(str)
+	if err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func DecodePkToString(bytes []byte, params *lib.DeSoParams, _ *lib.UtxoView) string {
+	return lib.PkToString(bytes, params)
+}
+
+func DecodePubKeyToUint64MapString(bytes []byte, params *lib.DeSoParams, utxoView *lib.UtxoView) string {
+	var decoded, err = lib.DeserializePubKeyToUint64Map(bytes)
+	if err != nil {
+		glog.Errorf("Error marshaling public key to uint64 map to string: %v", err)
+		// Fall back to default []byte -> string casting
+		return DecodeString(bytes, params, utxoView)
+	}
+	mapWithDecodedKeys := map[string]uint64{}
+	for k, v := range decoded {
+		mapWithDecodedKeys[lib.PkToString(k.ToBytes(), params)] = v
+	}
+	return fmt.Sprint(mapWithDecodedKeys)
+}
+
+func DecodeTransactionSpendingLimit(bytes []byte, params *lib.DeSoParams, utxoView *lib.UtxoView) string {
+	var transactionSpendingLimit *lib.TransactionSpendingLimit
+	if err := transactionSpendingLimit.FromBytes(bytes); err != nil {
+		glog.Errorf("Error decoding transaction spending limits: %v", err)
+		// Fall back to default []byte -> string casting
+		return DecodeString(bytes, params, utxoView)
+	}
+	response := TransactionSpendingLimitToResponse(transactionSpendingLimit, utxoView, params)
+	responseJSON, err := json.Marshal(response)
+	if err != nil {
+		glog.Errorf("Error marshaling transaction limit to string: %v", err)
+		// Fall back to default []byte -> string casting
+		return DecodeString(bytes, params, utxoView)
+	}
+	return string(responseJSON)
```

Comment:
What does this string look like?

Reply:
Same as above. This implementation isn't changed